lib/generators/honeycomb/honeycomb_generator.rb

```diff
@@ -22,7 +23,12 @@ def create_initializer_file
       <<-RUBY.strip_heredoc
         Honeycomb.configure do |config|
           config.write_key = #{write_key.inspect}
-          config.dataset = #{options['dataset'].inspect}
+          classic = config.write_key.nil? || config.write_key.length == 32
+          if classic
+            config.dataset = #{options['dataset'].inspect}
+          else
+            config.dataset = #{options['service_name'].inspect}
+          end
```

Comment:
This is code that gets added to the end-users code base the first time they install the beeline into a rails app. We could consider making this generator only do the not-classic behavior under the assumption that anyone adding the beeline to their app will be a new customer with an environment.